ethereumj-core/src/main/java/org/ethereum/vm/DataWord.java

```diff
@@ -22,77 +22,122 @@
 import org.ethereum.db.ByteArrayWrapper;
 import org.ethereum.util.ByteUtil;
 import org.ethereum.util.FastByteComparisons;
-import org.spongycastle.util.Arrays;
 import org.spongycastle.util.encoders.Hex;
 
 import java.math.BigInteger;
 import java.nio.ByteBuffer;
+import java.util.Arrays;
 
 import static org.ethereum.util.ByteUtil.toHexString;
 
 /**
  * DataWord is the 32-byte array representation of a 256-bit number
  * Calculations can be done on this word with other DataWords
+ * DataWord is immutable. Use one of `of` factories for instance creation.
  *
  * @author Roman Mandeleil
  * @since 01.06.2014
  */
-public class DataWord implements Comparable<DataWord> {
+public final class DataWord implements Comparable<DataWord> {
 
     /* Maximum value of the DataWord */
     public static final BigInteger _2_256 = BigInteger.valueOf(2).pow(256);
     public static final BigInteger MAX_VALUE = _2_256.subtract(BigInteger.ONE);
-    public static final DataWord ZERO = new DataWord(new byte[32]);      // don't push it in to the stack
-    public static final DataWord ZERO_EMPTY_ARRAY = new DataWord(new byte[0]);      // don't push it in to the stack
+    public static final DataWord ZERO = new DataWord(new byte[32]);
+    public static final DataWord ONE = DataWord.of((byte) 1);
 
     public static final long MEM_SIZE = 32 + 16 + 16;
 
-    private byte[] data = new byte[32];
+    private final byte[] data;
 
-    public DataWord() {
+    /**
+     * Unsafe private constructor
+     * Doesn't guarantee immutability if byte[] contents are changed later
+     * Use one of factory methods instead:
+     *  - {@link #of(byte[])}
+     *  - {@link #of(ByteArrayWrapper)}
+     *  - {@link #of(String)}
+     *  - {@link #of(long)}
+     *  - {@link #of(int)}
+     * @param data Byte Array[32] which is guaranteed to be immutable
+     */
+    private DataWord(byte[] data) {
+        if (data == null || data.length != 32) throw new RuntimeException("Input byte array should have 32 bytes in it!");
+        this.data = data;
     }
 
-    public DataWord(int num) {
-        this(ByteBuffer.allocate(4).putInt(num));
-    }
+    public static DataWord of(byte[] data) {
+        if (data == null || data.length == 0) {
+            return DataWord.ZERO;
+        }
+
+        boolean allExceptLastIsZero = true;
```

Comment:
You may use `8 * data.length - ByteUtil.numberOfLeadingZeros(data)`. And check the residual against `0` or `1`